File: examples/test_connection.ts
/**
 * Test your connection to inference-hub.
 *
 * Usage:
 *   npm install openai
 *   npx tsx test_connection.ts
 *
 * Set these environment variables before running:
 *   export INFERENCE_HUB_URL="http://192.168.1.100:4200/v1"
 *   export INFERENCE_HUB_KEY="mind-team-your-key-here"
 */

import OpenAI from "openai";

const BASE_URL = process.env.INFERENCE_HUB_URL ?? "http://localhost:4200/v1";
const API_KEY = process.env.INFERENCE_HUB_KEY ?? "YOUR_KEY_HERE";
const MODEL = process.env.INFERENCE_HUB_MODEL ?? "small";

const client = new OpenAI({ baseURL: BASE_URL, apiKey: API_KEY });

console.log(`Connecting to ${BASE_URL}...`);

// 1. List available models
const models = await client.models.list();
console.log(`Available models: ${models.data.map((m) => m.id).join(", ")}`);

// 2. Send a test message
console.log(`\nSending test message to model '${MODEL}'...`);
const response = await client.chat.completions.create({
  model: MODEL,
  messages: [{ role: "user", content: "What is 2+2? Reply in one sentence." }],
  max_tokens: 50,
});

console.log(`Response: ${response.choices[0].message.content}`);
console.log(`\nTokens used: ${response.usage?.total_tokens}`);
console.log("Connection successful!");

File: examples/test_connection.py
#!/usr/bin/env python3
"""
Test your connection to inference-hub.

Usage:
    pip install openai
    python test_connection.py

Set these environment variables before running:
    export INFERENCE_HUB_URL="http://192.168.1.100:4200/v1"
    export INFERENCE_HUB_KEY="your-api-key"

Or edit the defaults below.
"""

import os
import sys

try:
    from openai import OpenAI
except ImportError:
    print("Install the OpenAI SDK first:  pip install openai")
    sys.exit(1)

# -- Configuration --
BASE_URL = os.environ.get("INFERENCE_HUB_URL", "http://localhost:4200/v1")
API_KEY = os.environ.get("INFERENCE_HUB_KEY", "YOUR_KEY_HERE")
MODEL = os.environ.get("INFERENCE_HUB_MODEL", "small")

client = OpenAI(base_url=BASE_URL, api_key=API_KEY)

# 1. List available models
print(f"Connecting to {BASE_URL}...")
try:
    models = client.models.list()
    print(f"Available models: {[m.id for m in models.data]}")
except Exception as e:
    print(f"Connection failed: {e}")
    sys.exit(1)

# 2. Send a test message
print(f"\nSending test message to model '{MODEL}'...")
response = client.chat.completions.create(
    model=MODEL,
    messages=[{"role": "user", "content": "What is 2+2? Reply in one sentence."}],
    max_tokens=50,
)

print(f"Response: {response.choices[0].message.content}")
print(f"\nTokens used: {response.usage.total_tokens}")
print("Connection successful!")
